tensorflow/tools/ci_build/release/requirements_common.txt

```diff
@@ -1,21 +1,21 @@
 # To have reproducible builds, these dependencies should be pinned always.
 # Prefer pinning to the same version as in setup.py for now.
 # This will change in the future.
-absl-py ~= 1.0.0
+absl-py == 1.4.0
```

Comment:
Note that this is outside of range of previous spec. `~= 1.0.0` means `>= 1.0.0, < 1.1` but `1.4.0` is out of this range.

We might be lucky and things would work, we might be unlucky and get breakages if we use this high version.

Reply:
@mihaimaruseac Basically if the nightlies of branch cut day were green, we can be sure that the release build will be green too. correct?

Reply:
If not doing any pinning: green nightly build means green release build only under the assumption that all these packages have not seen a release _between branch cut and final release build_ (or until final patch cut, making this even harder to maintain). This is what caused breakage of RC0 last time even though nightly was green. A package got updated from under us.

If doing the pinning with the version that goes out of range then we have no guarantee that RC0 will be green. We have not run nightly with that package.

This is why I suggested **looking at nightly log for each wheel and seeing what package got installed and pinning to _that specific version_**.